HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,142 @@
+  # Add titles and axis labels
+  labs(title = "My Diamond Collection", subtitle = "Boxplot representing the number of diamonds in my diamond collection by \ntype of cut quality and clarity of diamond", x = "Diamond Cut", y = "Number of Diamonds") + 
+  # Center the titles
+  theme(plot.title = element_text(hjust = 0.5)) +
+  # Text annotation for diamonds with ideal cut
+  annotate("text", x = 4, y = 4500, label = "My Best Diamonds, \n of course") +
+  # Rectangular box to highlight diamonds with ideal cut 
+  annotate("rect", xmin = 4.5, xmax = 5.5, ymin = 0, ymax = 5000, alpha = 0.3)
+
+p1
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+p2 <- ggplot(iris, aes(x = Sepal.Length, y = Petal.Length, color = Species)) +
+  # Add the point layer to the plot, and specify shapes by species
+  geom_point(aes(shape = Species)) + 
+  # Wrap the plots for the three species to be three horizontally parallel graphs, and adjust the vertical scale for each plot 
+  facet_wrap(. ~Species, scales = "free_y") + 
+  # Add the regression lines, turn off the standard errors, change line color to black, and make lines to be thinner
+  geom_smooth(method = 'lm', se = FALSE, color = "black", size = 0.5)
+
+p2
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graoh 3 code}
+p3 <- ggplot(mpg, aes(x = displ, y = hwy)) +
+  geom_point() +
+  # Add an extra layer of points using the corvette dataset
+  geom_point(aes(x = displ, y = hwy), corvette, color = "blue") +
+  # Add the plot title
+  labs(title = "Corvettes are a bit of an outlier") + 
+  # Bolden the plot title
+  theme(plot.title = element_text(face = "bold")) + 
+  # Add annotations for the corvette data points
+  geom_text_repel(aes(x = displ, y = hwy, label = paste0("Corvette, ", year)), corvette)
+
+p3
+```
+
+There is a trick to getting the model and year to print off together. `paste()` is a useful function for this, also pasting together parts of file names and parts of urls together. 
+
+
+### Graph 4 
+```{r, echo = T}
+data(mpg)
+
+#hint for the coloring, colorbrewer and you can set palette colors and make your graphs colorblind friendly
+library(RColorBrewer)
+display.brewer.all(colorblindFriendly = T) #take a look at the colorblindfriendly options
+```
+
+The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
+
+
+
+Now using the above mpg dataset, make this graph
+
+
+```{r graph 4 code}
+p4 <- ggplot(mpg, aes(x = class, y = cty)) +
+  # Add the layer of jittered points and color by class
+  geom_point(aes (color = class), position  = position_jitter()) +
```

Comment:
It will be helpful to set width=0 for jittered points.

Reply:
I think you need to flip the coordinates to x-axis is Car class and y-axis is city mpg based on Robert's graph. Related to mingyu's comment, since the graph is supposed to be flipped, you may use height=0 in position_jitter() to make sure the jitters are on intrgers?